torch/csrc/api/include/torch/nn/functional/pooling.h

```diff
@@ -149,6 +149,52 @@ inline Tensor adaptive_avg_pool3d(const Tensor& input,
    return torch::adaptive_avg_pool3d(input, options.output_size());
 }
 
+// ============================================================================
+
+inline std::vector<int64_t> _unpool_output_size(const Tensor& input,
+  const IntArrayRef kernel_size, const IntArrayRef stride,
+  const IntArrayRef padding, const IntArrayRef output_size) {
```

Comment:
I think we should use `const IntArrayRef&` instead of `const IntArrayRef` for all arguments here.

Reply:
I think we should change `output_size` to `const c10::optional<IntArrayRef>&` to better match the Python version, because `output_size` being empty may not necessarily mean it's "None" (since a tensor with size `{}` is still a valid tensor, and it might be what the user wants).